### PR TITLE
Fix list_models bool parameters

### PR DESCRIPTION
**Pull request by Wauplin**
Fix https://github.com/huggingface/huggingface_hub/issues/1146.

With this PR, params `fetch_config` and `cardData` are not sent when value is `False` in `list_models` `list_datasets`. Also switched their type from `Optional[bool] = None` to `bool = False` for simplicity.

### CODE REVIEW

File: src/huggingface_hub/hf_api.py [hunk 1]
```diff
@@ -940,6 +940,11 @@ def _unpack_model_filter(self, model_filter: ModelFilter):
         query_dict["filter"] = tuple(filter_list)
         return query_dict
 
+    @_deprecate_arguments(
+        version="0.14",
+        deprecated_args={"cardData"},
+        custom_message="Use 'full' instead.",
+    )
```
[the remark this answers is not in the document]
Comment (LysandreJik) [reply src/huggingface_hub/hf_api.py:947]: Indeed, I tried it as well and it returns the same values. I'm not sure if this is intended though.

[the remark this answers is not in the document]
Comment (Wauplin) [reply src/huggingface_hub/hf_api.py:947]: In the codebase it was literally:

```py
if full is not None:
    if full:
        params.update({"full": True})
if cardData is not None:
    if cardData:
        params.update({"full": True})
```

And I tested in on the API, "cardData" doesn't seem to be a valid parameter for the `/datasets` endpoint. That's why I just removed the `cardData` param. I tend to think that it comes from a copy-paste of the arguments from `list_models` and not adapted correctly.

https://huggingface.co/api/datasets?limit=1 => base
https://huggingface.co/api/datasets?limit=1&cardData=1 => base
https://huggingface.co/api/datasets?limit=1&full=1 => full

**Review by LysandreJik — APPROVED** [review]
LGTM!

**Comment by Wauplin** [comment]
Thanks for the review and approval @LysandreJik, I'm merging :)

File: src/huggingface_hub/_login.py [hunk 2]
```diff
@@ -35,7 +35,7 @@
 logger = logging.get_logger(__name__)
 
 
-def login(token: Optional[str] = None, add_to_git_credential: bool = True) -> None:
+def login(token: Optional[str] = None, add_to_git_credential: bool = False) -> None:
```
Comment (julien-c) [inline src/huggingface_hub/_login.py:38]: i'm ok with switching that back to `True`

Comment (Wauplin) [reply src/huggingface_hub/_login.py:38]: Lol, let's keep as it is know (`False`). The user will be warned anyway.

**Pull request merged**